Add notice about repository deprecation in README.md (#374)
ros-tooling/cross_compile: README.md

> ## Notice: This repository is deprecated. It is outdated and has lack of support.

Reviewer: ```suggestion
## Notice: This repository is deprecated. It is outdated and will not be supported anymore.
```

> Recommended to use native cross-compilation which is up to 8 times faster.
> See details in ROS Discourse discussion
> https://discourse.ros.org/t/call-for-help-maintainership-of-the-ros-cross-compile-tool/26511 

Reviewer: ```suggestion
It is recommended to use a native cross-compilation method, which can be up to 8 times faster.
See details in this ROS Discourse discussion:
https://discourse.ros.org/t/call-for-help-maintainership-of-the-ros-cross-compile-tool/26511
```
